solidity/random-beacon/test/system/e2e.test.ts

```diff
@@ -0,0 +1,193 @@
+/* eslint-disable no-await-in-loop */
+
+import {
+  ethers,
+  waffle,
+  helpers,
+  getUnnamedAccounts,
+  getNamedAccounts,
+} from "hardhat"
+import { expect } from "chai"
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers"
+import {
+  constants,
+  dkgState,
+  params,
+  randomBeaconDeployment,
+} from "../fixtures"
+import type { RandomBeacon, RandomBeaconStub, TestToken } from "../../typechain"
+import {
+  genesis,
+  signAndSubmitCorrectDkgResult,
+  noMisbehaved,
+} from "../utils/dkg"
+import blsData from "../data/bls"
+import { registerOperators } from "../utils/operators"
+
+const ZERO_ADDRESS = ethers.constants.AddressZero
+
+const { to1e18 } = helpers.number
+const { mineBlocks, mineBlocksTo } = helpers.time
+const { keccak256 } = ethers.utils
+
+const fixture = async () => {
+  const contracts = await randomBeaconDeployment()
+
+  await registerOperators(
+    contracts.randomBeacon as RandomBeacon,
+    (await getUnnamedAccounts()).slice(1, 1 + constants.groupSize)
+  )
+
+  const randomBeacon = contracts.randomBeacon as RandomBeaconStub & RandomBeacon
+  const testToken = contracts.testToken as TestToken
+
+  return {
+    randomBeacon,
+    testToken,
+  }
+}
+
```

Comment:
Maybe some brief description of the test scenario?

Reply:
added d3e3969